### ISSUE TITLE
Mounted filesystem size always 8EB

### ISSUE THREAD

**Issue by netinfinity** [ovh/svfs#117]
### Context
- svfs version : 0.9.1
- storage provider : openstack swift
- product : - 

### Steps to reproduce this issue :
1. Add account quota in openstack swift
2. Check mounted filesystem with df -h

### Results you expected :
Expected to have mount of the size of the account quota as described in the code (fs.go - Statfs function)

### Results you observed :
Maximum size reported all the time.

### Additional information :
Need an information why account quota is not working on the mount and how to fix it or correctly mount it.

Thanks

**Comment by netinfinity** [comment]
Any help with this? Seems like svfs is not recognizing account quotas at all. Tested OS's were ubuntu 14.04 and 16.04.

**Comment by xlucas** [comment]
Hello,

As for now svfs only recognize account quotas once used with hubic.

For vanilla swift endpoints, it doesn't handle quotas yet, it is something that can be added in a further release, that's why the reported device size is set to the maximum underlying value.

This can be added to a future release for sure.

**Comment by netinfinity** [comment]
Thanks for the answer. Is there any workaround to limit filesystem size?

Thanks!

**Comment by xlucas** [comment]
None AFAIK but if you've set account quotas then obviously even if df doesn't display the actual filesystem size limit, trying to upload data beyond the account quota will fail.

**Comment by netinfinity** [comment]
Yeah, I know that. For me it's important that the size of the filesystem is the same as quota as it needs to be represented that way to a client.

Thank you for the help.

**Comment by netinfinity** [comment]
Sorry, one more question: How do I build svfs from source? It seems there is no documentation on how to build it.

Thanks!

**Comment by xlucas** [comment]
Hello, as a regular golang project : `go get github.com/ovh/svfs'.

**Comment by netinfinity** [comment]
Thanks again. Please do consider enabling the quota feature (either based on container or account quota) in future releases. It would be really helpful to have this feature, especially in some production cases.

Thank you!

**Comment by netinfinity** [comment]
Hello,

Any news on this feature. I have been trying to change /svfs/fs.go and /vendor/github.com/xlucas/swift/swift.go but no luck.

In swift.go I have changed info.Quota, _ = getInt64FromHeader(resp, "X-Account-Meta-Quota") to
info.Quota, _ = getInt64FromHeader(resp, "X-Account-Meta-Quota-Bytes"), which is the correct metadata information, and in fs.go I have changed Statfs reporting to if container is "" and account.quota > 0 then calculate filesystem size using the account quota information. So far I had no luck, it always reports as 8EB in size.
